webpack/components/extensions/HostDetails/Tabs/ErrataTab.js

```diff
@@ -104,38 +106,92 @@ export const ErrataTab = () => {
 
   if (!hostId) return <Skeleton />;
 
-  const rowActions = [
-    {
-      title: __('Apply via Katello agent'), disabled: true,
-    },
-    {
-      title: __('Apply via remote execution'), disabled: true,
-    },
-    {
-      title: __('Apply via customized remote execution'), disabled: true,
-    },
-  ];
+  const applyErratumViaRemoteExecution = id => dispatch(installErrata({
+    hostname,
+    search: errataSearchQuery(id),
+  }));
+
+  const applyViaRemoteExecution = () => {
+    dispatch(installErrata({
+      hostname, search: fetchBulkParams(),
+    }));
+
+    const params = { page: metadata.page, per_page: metadata.per_page, search: metadata.search };
+    dispatch(getInstallableErrata(
+      hostId,
+      { ...params, include_applicable: toggleGroupState === ALL },
+    ));
+  };
+
+  const bulkCustomizedRexUrl = () => errataInstallUrl({
+    hostname, search: (selectedCount > 0) ? fetchBulkParams() : '',
+  });
 
   const recalculateErrata = () => {
     setIsBulkActionOpen(false);
     dispatch(regenerateApplicability(hostId));
   };
 
+  const applyByKatelloAgent = () => {
+    const selected = fetchBulkParams();
+    setIsBulkActionOpen(false);
+    selectNone();
+    dispatch(applyViaKatelloAgent(hostId, { search: selected }));
+  };
+
+  const applyErratumViaKatelloAgent = id => dispatch(applyViaKatelloAgent(
+    hostId,
+    { errata_ids: [id] },
+  ));
+
+  const apply = () => (contentFacet.remoteExecutionByDefault ? applyViaRemoteExecution() :
```

Comment:
The Apply button uses different logic to decide whether to use Katello agent, compared to the dropdown items.  This uses
```js
 contentFacet.remoteExecutionByDefault
```
whereas the dropdowns use
```js
contentFacet.katelloAgentInstalled && contentFacet.katelloAgentEnabled
```

Was this intended? In my environment, the Apply button tries to use Katello agent, but I don't have any Katello agent actions in the dropdowns. This could be confusing.

Reply:
I guess we should just disable if `contentFacet.katelloAgentEnabled`  and ` contentFacet.remoteExecutionByDefault` are false

Reply:
Changed this condition a little. `(contentFacet.remoteExecutionByDefault || !katelloAgentAvailable)`
So if the thing cant be installed via katello agent it will still try rex (even if you have rexByDefault false).